std/variant.d

```diff
@@ -2204,6 +2247,13 @@ private auto visitImpl(bool Strict, VariantType, Handler...)(VariantType variant
                         result.indices[tidx] = dgidx;
                     }
                 }
+                else static if (isSomeFunction!(dg!T))
+                {
+                    assert(result.generalFuncIdx == -1 ||
+                           result.generalFuncIdx == dgidx,
+                           "Only one generic visitor function is allowed");
```

Comment:
I believe this code is actually run with CTFE, but I'll double check.

On May 9, 2016 5:17:11 PM EDT, Sebastian Wilzbach <notifications@github.com> wrote:
>> @@ -2204,6 +2247,13 @@ private auto visitImpl(bool Strict,
>VariantType, Handler...)(VariantType variant
>>                          result.indices[tidx] = dgidx;
>>                      }
>>                  }
>> +                else static if (isSomeFunction!(dg!T))
>> +                {
>> +                    assert(result.generalFuncIdx == -1 ||
>> +                           result.generalFuncIdx == dgidx,
>> +                           "Only one generic visitor function is
>allowed");
>
>Dumb question: wouldn't it be possible to throw the error at compile
>time? 
>
>---
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/dlang/phobos/pull/4035/files/c68ce5b47474034d91fef8ba0a69ae0523fefe31#r62574698

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.